Element::rendererIsEverNeeded() only has a default, hardcoded true implementation

twilco opened the pull request:
#### 8f8c86d4cb27f37d847f9e584c4cde775ce26bae
<pre>
Element::rendererIsEverNeeded() only has a default, hardcoded true implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=270726">https://bugs.webkit.org/show_bug.cgi?id=270726</a>
<a href="https://rdar.apple.com/124307290">rdar://124307290</a>

Reviewed by Aditya Keerthi.

After <a href="https://commits.webkit.org/274917@main">https://commits.webkit.org/274917@main</a>, only the default implementation
of Element::rendererIsEverNeeded() is left, making the function useless. This
patch removes it.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::rendererIsNeeded):
* Source/WebCore/dom/Element.h:
(WebCore::Element::rendererIsEverNeeded): Deleted.
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::rendererIsEverNeeded): Deleted.
* Source/WebCore/html/HTMLElement.h:
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolveElement):

Canonical link: <a href="https://commits.webkit.org/275876@main">https://commits.webkit.org/275876@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b36abd5ab56e0042cbfea5e412a195d9cb0df0e2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43122 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22146 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45522 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45746 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39245 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45426 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25887 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19569 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35643 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43695 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19192 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37145 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16641 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16778 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38208 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1176 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39315 "Found 1 new API test failure: TestWebKitAPI.DataDetectorTests.LoadWKWebViewWithDataDetectorTypePhoneNumber (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38531 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47289 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18036 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14820 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42433 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19573 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41092 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9600 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19751 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19205 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
LGTM – this method became useless after https://commits.webkit.org/274917@main.

It might even be nice to mention that in the ChangeLog just to make it easier to keep track of history.

twilco committed:
Element::rendererIsEverNeeded() only has a default, hardcoded true implementation
https://bugs.webkit.org/show_bug.cgi?id=270726
rdar://124307290

Reviewed by Aditya Keerthi.

After https://commits.webkit.org/274917@main, only the default implementation
of Element::rendererIsEverNeeded() is left, making the function useless. This
patch removes it.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::rendererIsNeeded):
* Source/WebCore/dom/Element.h:
(WebCore::Element::rendererIsEverNeeded): Deleted.
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::rendererIsEverNeeded): Deleted.
* Source/WebCore/html/HTMLElement.h:
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::resolveElement):

Canonical link: https://commits.webkit.org/275876@main
